js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -876,8 +882,13 @@ const EditingController = modules.ViewController.inherit((function() {
             when(that._initNewRow(param, parentKey)).done(() => {
                 if(that._allowRowAdding()) {
                     that._addRowCore(param.data, parentKey, oldEditRowIndex);
+                } else {
+                    deferred.reject('cancel');
                 }
-            });
+                deferred.resolve();
```

Comment:
Move this resolve to condition block with 'that._allowRowAdding()'

Reply:
Done